src/modules/ie-objects/helpers/limit-access-to-object-details.ts

```diff
@@ -32,71 +37,130 @@ export const limitAccessToObjectDetails = (
 		accessibleVisitorSpaceIds: string[];
 	}
 ): Partial<IeObject> => {
-	// Step 1 - Determine Licenses
-	// ---------------------------------------------------
-	let userGroup = isNil(userInfo?.userId)
-		? IE_OBJECT_EXTRA_USER_GROUPS[IeObjectExtraUserGroupType.ANONYMOUS]
-		: userInfo.groupId;
+	let ieObjectLicenses: IeObjectLicense[] = [...ieObject.licenses];
+
+	const hasIntraCPLicenses = intersection(ieObjectLicenses, IE_OBJECT_INTRA_CP_LICENSES);
+	const hasPublicLicenses = intersection(ieObjectLicenses, IE_OBJECT_PUBLIC_LICENSES);
```

Comment:
bad naming for these variables. "has" indicates that this is a boolean
A better name might be:
objectIntraCpLicenses
objectPublicLicenses